src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -715,64 +713,75 @@ _nns_edge_message_handler (void *thread_data)
 
   conn->running = true;
   while (conn->running) {
-    nns_edge_data_h data_h;
-    unsigned int i;
+    struct pollfd poll_fd;
 
     /* Validate edge handle */
     if (!NNS_EDGE_MAGIC_IS_VALID (eh)) {
       nns_edge_loge ("The edge handle is invalid, it would be expired.");
       break;
     }
 
-    /** Receive data from the client */
-    _nns_edge_cmd_init (&cmd, _NNS_EDGE_CMD_ERROR, client_id);
-    ret = _nns_edge_cmd_receive (conn, &cmd);
-    if (ret != NNS_EDGE_ERROR_NONE) {
-      nns_edge_loge ("Failed to receive data from the connected node.");
-      remove_connection = true;
-      break;
-    }
+    poll_fd.fd = conn->sockfd;
+    poll_fd.events = POLLIN | POLLHUP | POLLERR;
+    poll_fd.revents = 0;
 
-    if (cmd.info.cmd == _NNS_EDGE_CMD_ERROR) {
-      nns_edge_loge ("Received error, stop msg thread.");
-      _nns_edge_cmd_clear (&cmd);
-      remove_connection = true;
-      break;
-    }
+    /* 10 milliseconds */
+    if (poll (&poll_fd, 1, 10) > 0) {
```

Comment:
Don't need to handle error case except timeout?

Reply:
waiting for new data here, timeout is no data received.

Reply:
And, _nns_edge_cmd_receive() checks connection state internally.